apps/deposit-morpho/src/components/VaultForm.tsx

```diff
@@ -150,8 +151,31 @@ export function VaultForm({ vault }: { vault: Vault }) {
       />
       <Info content={<InfoContent />} />
       <ButtonPrimary type="submit" className="mb-0">
-        Add Hook
+        <ButtonText context={context} />
       </ButtonPrimary>
     </div>
   );
 }
+
+const ButtonText = ({
+  context,
+  errorMessage,
+  isLoading,
+}: {
+  context: HookDappContextAdjusted;
+  errorMessage?: string | undefined;
+  isLoading?: boolean;
+}) => {
+  if (errorMessage) return <span>{errorMessage}</span>;
+
+  if (isLoading) return <span>Loading...</span>;
+
+  if (context?.hookToEdit && context?.isPreHook)
+    return <span>Update Pre-hook</span>;
+  if (context?.hookToEdit && !context?.isPreHook)
+    return <span>Update Post-hook</span>;
+  if (!context?.hookToEdit && context?.isPreHook)
+    return <span>Add Pre-hook</span>;
+  if (!context?.hookToEdit && !context?.isPreHook)
+    return <span>Add Post-hook</span>;
+};
```

Comment:
Why not wrap this in a useMemo instead of a component? It seems to me that is only text diffs based on some field conditions 🤔